File: dabbit.Base/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace dabbit.Base
{
    public class Message
    {
        public string[] Parts { get; internal set; }
        public string MessageLine { get; internal set; }
        public string Command { get; internal set; }
        public string RawLine { get; internal set; }

        public SourceEntity From { get; internal set; }
        public DateTime Timestamp { get; internal set; }

    }
    public class SourceEntity
    {
        public SourceEntityType Type { get { return this.fromType; } }
        public string[] Parts { get { return this.parts; } }

        public SourceEntity(string[] parts, SourceEntityType sourceType)
        {
            this.fromType = sourceType;
            this.parts = parts;
        }

        private SourceEntityType fromType;
        private string[] parts;
    }
    public enum SourceEntityType
    {
        Server,
        Client,
        BNC,
        Dabbit,
        Channel
    }
}

File: dabbit/Base/Message.js
var System = new require("all")("System");

function Message() {
    // Indicates object inheritance.
    System.Object.call(this);
    
    this.Parts = [];
    this.MessageLine = String.Empty;
    this.Command = String.Empty;
    this.RawLine = String.Empty;

    this.From = {};
    this.Timestamp = {};
}
System.Javascript.Inherit(System.Object, Message);

module.exports = Message;

